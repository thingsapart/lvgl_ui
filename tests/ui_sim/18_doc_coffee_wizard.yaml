# TICKS: 6
- type: data-binding
  state:
    - setup_step: 0 # 0=Welcome, 1=Strength, 2=Size, 3=Confirm
  actions:
    - next_step:
        inc: { setup_step: 1 }
    - prev_step:
        dec: { setup_step: 1 }
    - finish_setup:
        set: { setup_step: 0 } # Reset after completion
  updates:
    # Clamp step to valid range
    - set: { setup_step: [clamp, setup_step, 0, 3] }
  schedule:
    - { tick: 1, action: next_step }
    - { tick: 2, action: next_step }
    - { tick: 3, action: next_step }
    - { tick: 4, action: next_step } # Should be clamped
    - { tick: 5, action: prev_step }
    - { tick: 6, action: finish_setup }
