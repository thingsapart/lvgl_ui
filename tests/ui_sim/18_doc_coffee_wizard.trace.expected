--- UI-Sim Trace Start ---
STATE_SET: setup_step = 0.000 (old: null)
NOTIFY: setup_step = 0.000

--- TICK 1 ---
ACTION: next_step value=null
STATE_SET: setup_step = 1.000 (old: 0.000)
NOTIFY: setup_step = 1.000

--- TICK 2 ---
ACTION: next_step value=null
STATE_SET: setup_step = 2.000 (old: 1.000)
NOTIFY: setup_step = 2.000

--- TICK 3 ---
ACTION: next_step value=null
STATE_SET: setup_step = 3.000 (old: 2.000)
NOTIFY: setup_step = 3.000

--- TICK 4 ---
ACTION: next_step value=null
STATE_SET: setup_step = 3.000 (old: 3.000)

--- TICK 5 ---
ACTION: prev_step value=null
STATE_SET: setup_step = 2.000 (old: 3.000)
NOTIFY: setup_step = 2.000

--- TICK 6 ---
ACTION: finish_setup value=null
STATE_SET: setup_step = 0.000 (old: 2.000)
NOTIFY: setup_step = 0.000

--- UI-Sim Trace End ---
